test/src/sdk-harness/test_projects/storage_vec/utils/mod.rs

```diff
@@ -0,0 +1,62 @@
+use fuels::{prelude::*, tx::ContractId};
+// Load abi from json
+abigen!(MyContract, "test_artifacts/storage_vec/svec_u8/out/debug/svec_u8-abi.json");
+
+
+pub mod setup {
+    use super::*;
+
+    pub async fn get_contract_instance() -> (MyContract, ContractId) {
+        // Launch a local network and deploy the contract
+        let wallet = launch_provider_and_get_single_wallet().await;
+    
+        let id = Contract::deploy("test_artifacts/storage_vec/svec_u8/out/debug/svec_u8.bin", &wallet, TxParameters::default())
+            .await
+            .unwrap();
+    
+        let instance = MyContract::new(id.to_string(), wallet);
+    
+        (instance, id)
+    }
+    
+}
+
+pub mod wrappers {
+    use super::*;
+
+    pub async fn push(instance: &MyContract, value: u8) {
```

Comment:
If this is meant to be a wrapper (abi_calls - think that is a better name) then perhaps you could try using generics once you rename the functions? If each contract has the exact same ABI names then maybe you could write a single module to use across all of your contracts instead of having to pass in specific types each time.

That is assuming that you can use generics in your function parameters - which I think you can.

Reply:
Perhaps. I do not know if you can do some magic where you put all of the generated types into a vec and iterate over them. If they all have the same signatures then you should be able to call them with 1 name. I might be mistaken and it could be that Rust doesn't allow this.